scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -590,19 +586,40 @@ def start( self ):
             else:
                 self.__k8s = KubernetesApi( ca_file=None )
 
-            self.__k8s_filter = self._build_k8s_filter()
-            self._logger.log( scalyr_logging.DEBUG_LEVEL_1, "k8s filter for pod '%s' is '%s'" % (self.__k8s.get_pod_name(), self.__k8s_filter) )
-
             self.__client = DockerClient( base_url=('unix:/%s'%self.__socket_file), version=self.__docker_api_version )
 
             self.container_id = self.__get_scalyr_container_id( self.__client, self.__name )
 
             # create the k8s cache
             self.k8s_cache = KubernetesCache( self.__k8s, self._logger,
                 cache_expiry_secs=self.__k8s_cache_expiry_secs,
-                max_cache_misses=self.__k8s_max_cache_misses,
-                cache_miss_interval=self.__k8s_cache_miss_interval,
-                filter=self.__k8s_filter )
+                cache_purge_secs=self.__k8s_cache_purge_secs,
+                namespaces_to_ignore=self.__namespaces_to_ignore )
+
+
+            delay = 0.25
+            backoff = 1.5
+            message_delay = 6
+            kill_delay = 20
+            start_time = time.time()
+
+            while not self.k8s_cache.is_initialized():
```

Comment:
Ok, so the cache predated Starbuck support?  They've always had to have a service account?

Reply:
I'm not sure if it predated it, but I think it was.  I created it not for Starbuck, but to avoid needing to make repeated queries to the API for the the same object during the same loop.

Regardless of how it was created, it is now definitely needed independently of Starbuck.

Regarding the service account, they've haven't *always* needed one, because k8s used to have a weaker security model.  Customers have always needed one since the introduction of RBAC.

So yes, a service account is required for anyone using the k8s monitor regardless of whether they use starbuck or not.  It's just that for starbuck we need to access slightly more endpoints than we would without starbuck.  I don't think it's worth splitting the service account in to two separate ones depending on whether starbuck is enabled or not.